Display a toast when establishing WalletConnect connection

DRadmir opened the pull request:
## Summary
Replaces loading indicators on connection buttons with a persistent toast bar for connection feedback. Removes loading state from ButtonListItem and related logic from ConnectionsViewModel. Updates ToastModifier to support custom duration and tap-to-dismiss options, and applies these changes to the ConnectionsScene.

Closes #1207

## Screenshot

<img width="320" alt="Simulator Screenshot - iPhone 17 Pro - 2025-09-29 at 16 38 55" src="https://github.com/user-attachments/assets/095ac0e9-a2fb-4a35-93fa-cb466de5242d" />

DRadmir committed:
Add per-button loading to Connections scene

gemcoder21 commented on Features/WalletConnector/Sources/WalletConnector/Scenes/ConnectionsScene.swift:
```diff
@@ -22,18 +22,15 @@ public struct ConnectionsScene: View {
                 ButtonListItem(
                     title: model.pasteButtonTitle,
                     image: Images.System.paste,
+                    isLoading: model.loadingButton == .paste,
```
```suggestion
                    isLoading: model.isLoadingPasteButton,
```

DRadmir committed:
Merge branch 'main' of github.com:gemwalletcom/gem-ios into 1207-refactoring-loading-state-for-wc

DRadmir committed:
Merge branch 'main' of github.com:gemwalletcom/gem-ios into 1207-refactoring-loading-state-for-wc

DRadmir committed:
Refactor connection UI to use toast for feedback

Replaces loading indicators on connection buttons with a persistent toast bar for connection feedback. Removes loading state from ButtonListItem and related logic from ConnectionsViewModel. Updates ToastModifier to support custom duration and tap-to-dismiss options, and applies these changes to the ConnectionsScene.

gemdev111 commented on Features/WalletConnector/Sources/WalletConnector/Scenes/ConnectionsScene.swift:
```diff
@@ -72,11 +68,20 @@ public struct ConnectionsScene: View {
         .onChange(of: model.walletConnectorPresenter?.isPresentingSheet?.id) { _, newValue in
             if newValue != nil {
-                model.stopLoading()
+                model.hideConnectionBar()
```
can we handle onchange in scene view model like we do `.onChange(of: model.filterModel, model.onChangeFilterModel)` but in this case onChangeSheetType or relevant

gemcoder21 commented on Packages/Components/Sources/ViewModifiers/ToastModifier.swift:
```diff
@@ -50,11 +66,29 @@ private struct OptionalMessageToastModifier: ViewModifier {
-        modifier(OptionalMessageToastModifier(message: message))
+    func toast(
+        message: Binding<ToastMessage?>,
+        duration: Double = 2,
```
add 2 to const, as it's used above

DRadmir committed:
Refactor sheet change handler and toast duration

Simplifies ConnectionsScene sheet change handler by passing method reference. Centralizes toast duration in ToastModifier with a static property for consistency.
